composer/callbacks/image_monitor.py

```diff
@@ -0,0 +1,194 @@
+from typing import Any, Callable, Tuple, Union
+
+import torch
+
+from composer.core import Callback, State, Time, TimeUnit
+from composer.loggers import Logger
+from composer.utils.import_helpers import MissingConditionalImportError
+
+__all__ = ['ImageMonitor']
+
+
+class ImageMonitor(Callback):
+    """Logs image inputs and optionally outputs.
+
+    This callback triggers at a user defined interval, and logs a sample of input (optionally also segmentation masks)
+    images under the ``Images/Train`` and ``Image/Eval`` keys.
+
+    Example:
+    .. doctest::
+
+        >>> from composer import Trainer
+        >>> from composer.callbacks import ImageMonitor
+        >>> # constructing trainer object with this callback
+        >>> trainer = Trainer(
+        ...     model=model,
+        ...     train_dataloader=train_dataloader,
+        ...     eval_dataloader=eval_dataloader,
+        ...     optimizers=optimizer,
+        ...     max_duration="1ep",
+        ...     callbacks=[ImageMonitor()],
+        ... )
+
+    The images are logged by the :class:`.Logger` to the following key(s) as described
+    below.
+
+    +---------------------------------------------+---------------------------------------+
+    | Key                                         | Logged data                           |
+    +=============================================+=======================================+
+    | ``Images/Train``                            |   Sampled examples of train images    |
+    +---------------------------------------------+---------------------------------------+
+    | ``Images/Eval``                             |   Sampled examples of eval images     |
+    +---------------------------------------------+---------------------------------------+
+
+    Args:
+        interval (str, optional): Time string specifying how often to log train images. For example, ``interval='1ep'``
+            means images are logged once every epoch, while ``interval='100ba'`` means images are logged once every 100
+            batches. Eval images are logged once at the start of each eval. Default: ``"100ba"``.
+        mode (str, optional): How to log the image labels. Valid values are ``"input"`` (input only)
+            and "segmentation" which also saves segmentation masks. Default: ``"input"``.
+        num_images (int, optional): Number of images to log. Should be less than or equal to than the microbatch size.
+            If there are not enough images in the microbatch, all the images in the microbatch will be logged.
+            Default: ``8``.
+        shift_targets (bool): For use when ``mode="segmentation"``. Optionally shifts the target classes up by one to
+            allow for ignored/background classes with negative values. If ``shift_targets=True``, all class ids will be
+            incremented by 1, and any pixels with negative target indices will be assigned a target index of zero.
+            Default: ``False``.
```

Comment:
I agree with this. It should be changed to this behavior now, and as a bonus you no longer need to specify a `shift_targets` option because targets are never shifted.